cmd/nodepool/core/create.go

```diff
@@ -0,0 +1,116 @@
+package core
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/cmd/util"
+	hyperapi "github.com/openshift/hypershift/support/api"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type CreateNodePoolOptions struct {
+	Name         string
+	Namespace    string
+	ClusterName  string
+	NodeCount    int32
+	ReleaseImage string
+	Render       bool
+}
+
+// ApplyPlatformSpecifics can be used to create platform specific values as well as enriching the fixure with additional values
+type CreateNodePoolPlatformOptions interface {
+	UpdatePlatformParams(ctx context.Context, nodePool *hyperv1.NodePool, hcluster *hyperv1.HostedCluster, client crclient.Client) error
+	PlatformType() hyperv1.PlatformType
+}
+
+func (o *CreateNodePoolOptions) CreateNodePool(ctx context.Context, platformOpts CreateNodePoolPlatformOptions) error {
+	client := util.GetClientOrDie()
+
+	hcluster := &hyperv1.HostedCluster{}
+	err := client.Get(ctx, types.NamespacedName{Namespace: o.Namespace, Name: o.ClusterName}, hcluster)
+	if err != nil {
+		return fmt.Errorf("failed to get HostedCluster %s/%s: %w", o.Namespace, o.Name, err)
+	}
+
+	if platformOpts != nil {
```

Comment:
do we need this check? when would this be nil?

Reply:
You right, it never should be nil, removed the validation